TU20Bot/Commands/NameMatch.cs

```diff
@@ -0,0 +1,220 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Discord;
+using Discord.Commands;
+using Discord.WebSocket;
+using Org.BouncyCastle.Math.EC.Rfc7748;
+using TU20Bot.Configuration;
+
+namespace TU20Bot.Commands {
+    public class NameMatch : ModuleBase<SocketCommandContext> {
+
+        // The max limit of chars set by discord
+        private const int messageLimit = 2000;
+        private Config _config;
+
+        public enum MatchLevel {
+            NoMatch,
+            CloseMatch,
+            CompleteMatch
+        }
+
+        public class MatchResult {
+            public MatchLevel level { get; set; }
+            public List<string> noSpacesMatch { get; set; }
+            public List<string> lastNameMatch { get; set; }
+            public SocketGuildUser user { get; set; }
+            public string fullName { get; set; }
+        }
+
+        [Command("lsuser-assignrole")]
+        [Summary("matches guild member names with a specified list and assigns roles to the matched names")]
+        public async Task matchNames(ulong roleId) {
+
+            var users = Context.Guild.Users;
+
+            _config = ((Client)Context.Client).config;
+
+            await sendSplitMessage(await nameMatching(_config.userDataCsv, users, roleId), "\n");
+        }
```

Comment:
Is this only for debugging? Can we remove it?

One thing I am worried about is regular users running commands they should probably not be able to access.

Reply:
Great point! Unlike the email verification system, this shouldn't be a feature that is triggered through discord. Let's disable the command for now and we will work on triggering it from the frontend later.